File: src/main/kotlin/domain/account/data/IAccountInfoRepo.java
package domain.account.data;

import org.elegantobjects.jpages.LibraryApp.common.util.Result;
import org.elegantobjects.jpages.LibraryApp.common.util.uuid2.UUID2;
import org.elegantobjects.jpages.LibraryApp.domain.account.Account;
import org.elegantobjects.jpages.LibraryApp.domain.common.data.repo.IRepo;

/**
 * IAccountInfoRepo is an interface for the AccountInfoRepo class.
 *
 * @author Chris Athanas (realityexpanderdev@gmail.com)
 * @since 0.11
 */

public interface IAccountInfoRepo extends IRepo {
    Result<AccountInfo> fetchAccountInfo(UUID2<Account> id);
    Result<AccountInfo> updateAccountInfo(AccountInfo accountInfo);
    Result<AccountInfo> upsertAccountInfo(AccountInfo accountInfo);
}

File: src/main/kotlin/domain/account/data/IAccountInfoRepo.kt
package domain.account.data

import common.uuid2.UUID2
import domain.account.Account
import domain.common.data.repo.IRepo

/**
 * IAccountInfoRepo is an interface for the AccountInfoInMemoryRepo class.
 *
 * @author Chris Athanas (realityexpanderdev@gmail.com)
 * @since 0.12 Kotlin conversion
 */

interface IAccountInfoRepo : IRepo {
    suspend fun fetchAccountInfo(id: UUID2<Account>): Result<AccountInfo>
    suspend fun fetchAllAccountInfo(): Result<List<AccountInfo>>
    suspend fun updateAccountInfo(accountInfo: AccountInfo): Result<AccountInfo>
    suspend fun upsertAccountInfo(accountInfo: AccountInfo): Result<AccountInfo>
    suspend fun deleteAccountInfo(accountInfo: AccountInfo): Result<Unit>
    suspend fun deleteDatabase(): Result<Unit>
}
